W 16422273 api ext auto-completion props duke (#347)
mulesoft/docs-code-builder: modules/ROOT/pages/_partials/acb-api-design.adoc

----

// end::apid-best-practices-raml[]
//
//
// tag::apid-extension-props[]
. Open the *Extensions* panel from the activity bar by clicking the image:anypoint-code-builder::icon-extensions.png["",18,18] (Extensions) icon.
. Right-click the extension *Anypoint Code Builder - API Extension*, and select *Extensions Settings*.
+

> This action opens the *Settings* tab to the configuration for the extension, *@ext:salesforce.mule-dx-api-component*.

Author: TODO: IS "@ext:salesforce.mule-dx-api-component" final? And where in the Settings tab will this setting appear? At the end of the Mule settings? 

@mansillafederico & @lbauret

Reviewer: `@ext:salesforce.mule-dx-api-component` is auto completed by vscode, I think it a way to filter among other extensions. It appears at the top of the tab because is already filtered by 'mule-dx-api-component'.

> . In the *Settings* tab, enable the API design editor to suggest auto-completion options. 
> +
> This setting is enabled by default. 

Author: ```suggestion
. In the *Settings* tab, enable the API design editor to suggest auto-completion options. 
+
The *Mule > Editor > Auto-completion* setting is disabled by default. 
```

> The *Mule > Editor > Auto-completion* setting is disabled by default. 

Author: ```suggestion
See xref:ref-mule-settings.adoc#setting-auto-complete[Mule > Editor > Auto-completion].
```

> See xref:ref-mule-settings.adoc#setting-auto-complete[Mule > Editor > Auto-completion].

Reviewer 2: I would prefer `Mule › Editor › Auto-completion: Enabled` here

Author: @ignaciovazqueza I left a message. I think this setting name should change.

Author: to `Mule › Editor › Auto-completion` or `Mule › Editor › Auto-completion enablement` or similar but not have a name that implies that the setting is enabled.

> See xref:ref-mule-settings.adoc#setting-auto-complete[Mule > Editor > Auto-completion].

Author: ```suggestion
See xref:ref-mule-settings.adoc#setting-auto-complete[Mule > Editor > Auto-completion: Enablement].
```

. Optionally, set a time in milliseconds to wait before listing auto-completion options.
+

> The default is `500` milliseconds. 

Reviewer: We changed the default value to `2000` milliseconds
cc. @lbauret @dukesphere

Author: updated

> The default is `500` milliseconds. 

Author: ```suggestion
The default is `2000` milliseconds. 
```

> The default is `2000` milliseconds. 

Reviewer: The default delay was moved to `1500` milliseconds.
cc. @dukesphere

Author: updated

> The default is `2000` milliseconds. 

Author: ```suggestion
The default is `1500` milliseconds. 
```

> The default is `1500` milliseconds. 

Author: ```suggestion
The default for *Mule > Editor > Auto-completion: Trigger Delay* is `1500` milliseconds. 
```

> The default for *Mule > Editor > Auto-completion: Trigger Delay* is `1500` milliseconds. 

Author: ```suggestion
See xref:ref-mule-settings.adoc#setting-auto-complete[Mule > Editor > Auto-completion: Trigger Delay]. 
```
